router/internal/httpclient/cancelable_transport_test.go

```
@@ -0,0 +1,285 @@
+	transport := NewCancelableRoundTripper(roundTripperFunc(func(*http.Request) (*http.Response, error) {
+		<-release
+		return nil, nil
+	}), supervisedOptions(observer, 1))
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Millisecond)
+	defer cancel()
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://example.com", nil)
+	require.NoError(t, err)
+
+	started := time.Now()
+	resp, err := transport.RoundTrip(req)
```

golangci-lint (2.12.2)

[error] response body must be closed

(bodyclose)